pandas/_libs/meson.build

```diff
@@ -61,12 +61,12 @@ subdir('tslibs')
 libs_sources = {
     # Dict of extension name -> dict of {sources, include_dirs, and deps}
     # numpy include dir is implicitly included
-    'algos': {'sources': ['algos.pyx', _algos_common_helper, _algos_take_helper, _khash_primitive_helper]},
-    'arrays': {'sources': ['arrays.pyx']},
+    'algos': {'sources': ['algos.pyx', _algos_common_helper, _algos_take_helper], 'deps': _khash_primitive_helper_dep},
+    'arrays': {'sources': ['arrays.pyx'], 'deps': _khash_primitive_helper_dep},
```

Comment:
seems weird that this would be needed here

Reply:
Agreed with Brock, doesn't arrays not cimport anything from pandas?

Reply:
Right you are! 👍 I reckon it hit that in 'earlier' tries, and the proper fix was in index? 🤷

Reply:
Well, it does seem to build, but breaks the unit tests?

Reply:
No, that's unrelated. 

#55804 should fix it.